tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,433 @@
+               "H3F3A", "IGF1R", "KDM6A", "KIAA1549", "KIT", 
+               "KMT2C", "KMT2D", "KRAS", "KSR2", "MAML2", "MAMLD1", 
+               "MET", "MIR512-2", "MYB", "MYCN", "NF1", "NF2", 
+               "NTRK2", "NTRK3", "PDGFRA", "PIK3CA", "PIK3R1", 
+               "PPM1D", "PRKAR1A", "PTCH1", "PTEN", "PTPN11", 
+               "QKI", "RAF1", "RELA", "ROS1", "SETD2", "SMARCA4", 
+               "TACC1", "TCF4", "TERT", "TP53", "YAP1", "ZIC1")
+
+select_cols <- c("Tumor_Sample_Barcode", "Hugo_Symbol", "Variant_Classification")
+
+
+# Combine all alterations:
+alterations_df <- list(maf_df, fusion_df, cnv_df) %>%
+  # keeping it interesting
+  purrr::map(
+    dplyr::select,
+    select_cols
+  ) %>%
+  dplyr::bind_rows() %>%
+  # rename Tumor_Sample_Barcode to sample_id
+  dplyr::rename(sample_id = Tumor_Sample_Barcode) %>%
+  # filter to only genes in gene_list
+  dplyr::filter(Hugo_Symbol %in% gene_list) %>%
+  # some wrangling to help with dev:
+  dplyr::arrange(Hugo_Symbol) %>%
+  tibble::as_tibble()
+
+# Ensure we still have all the genes at least _somewhere_ in the data frame
+if (length(unique(alterations_df$Hugo_Symbol)) != length(gene_list)) {
+  stop("Some genes are missing from the alteration df.")
+}
+
+# Convert to presence/absence of _all combinations_ of genes and variants for each sample_id
+alterations_presence_df <- alterations_df %>%
+  # If the row is here already, then the alteration is present
+  dplyr::mutate(present = TRUE) %>% 
+  # Fill in all combinatins of gene/variant for each sample, 
+  #  filling in FALSE for present for any new rows 
+  tidyr::complete(sample_id, Hugo_Symbol, Variant_Classification, 
+                  fill = list(present = FALSE)) %>%
+  dplyr::distinct()
+
+  
+# At this point, alterations_presence_df is a Y/N tibble, but it contains only `sample_id`, not DNA and RNA biospecimen ids.
+# The final step is to add that information in, but we need to handle different groups separately:
+# - First, the fully non-ambiguous samples
+# - Second, the ambiguous samples: where DNA and RNA ids cannot be reliably linked, OR 
+#   where there are multiple same-modality experiments (since we are working with all samples, not `primary-only`)
+
+## Combine alterations_presence_df with biospecimen sample ids ----------------------
+  
+# First, let's "combine" the ambiguous ids: 
+# For multiple DNA or RNA for a given sample_id, convert to a semi-colon separated single value
+# For example, two DNA ids on separate rows named BS_1 and BS_2 would be collapsed into `BS_1; BS2`
+merged_ambiguous_ids <- tumor_sample_ids_df %>%
+  # get only samples of interest
+  dplyr::filter(ambiguous | multiple_exp_strategy) %>%
+  # for each sample_id, combine biospecimen ids by modality to have 
+  # at most one "value" (semi-colon separated list) for each modality 
+  dplyr::group_by(sample_id, modality) %>%
+  dplyr::summarize(Kids_First_Biospecimen_ID = paste(Kids_First_Biospecimen_ID, collapse="; ")) %>%
```

Comment:
I'd leave out the space: easier to unparse later.

Also, just in case, adding unique and sort here: 
```suggestion
  dplyr::summarize(
    Kids_First_Biospecimen_ID = paste(
      sort(unique(Kids_First_Biospecimen_ID)),
      collapse=";")
  ) %>%
```

Reply:
Further thought: if you don't filter to just ambiguous, you can put that indicator here as well, and save having to have two flows, I think. 

```suggestion
    dplyr::summarize(
    Kids_First_Biospecimen_ID = paste(
      sort(unique(Kids_First_Biospecimen_ID)),
      collapse=";"),
    ambiguous_biospecimen_mapping = dplyr::n()>1
  ) %>%
```